core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -132,6 +132,29 @@ public class RESTSessionCatalog extends BaseViewSessionCatalog
           .addAll(TOKEN_PREFERENCE_ORDER)
           .build();
 
+  private static final Set<Endpoint> DEFAULT_ENDPOINTS =
+      ImmutableSet.<Endpoint>builder()
+          .add(ResourcePaths.V1_LIST_NAMESPACES)
+          .add(ResourcePaths.V1_LOAD_NAMESPACE)
+          .add(ResourcePaths.V1_CREATE_NAMESPACE)
+          .add(ResourcePaths.V1_UPDATE_NAMESPACE)
+          .add(ResourcePaths.V1_DELETE_NAMESPACE)
+          .add(ResourcePaths.V1_LIST_TABLES)
+          .add(ResourcePaths.V1_LOAD_TABLE)
+          .add(ResourcePaths.V1_CREATE_TABLE)
+          .add(ResourcePaths.V1_UPDATE_TABLE)
+          .add(ResourcePaths.V1_DELETE_TABLE)
+          .add(ResourcePaths.V1_RENAME_TABLE)
+          .add(ResourcePaths.V1_REGISTER_TABLE)
+          .add(ResourcePaths.V1_REPORT_METRICS)
+          .add(ResourcePaths.V1_LIST_VIEWS)
+          .add(ResourcePaths.V1_LOAD_VIEW)
+          .add(ResourcePaths.V1_CREATE_VIEW)
+          .add(ResourcePaths.V1_UPDATE_VIEW)
+          .add(ResourcePaths.V1_DELETE_VIEW)
+          .add(ResourcePaths.V1_RENAME_VIEW)
```

Comment:
By including view routes, this fails to solve the problem that we set out to address. That basically means that we have to add new handling to see if `endpoints` was sent to the locations with that ambiguous error handling, making the code more complicated rather than fixing it.